src/params.f90

```diff
@@ -52,23 +50,378 @@ module params
   real(r64), parameter :: Ryd2eV = 0.5_r64*Hartree2eV !Rydberg to eV
   real(r64), parameter :: Ryd2meV = Ryd2eV*1.0e3_r64 !Rydberg to meV
   real(r64), parameter :: Ryd2amu = 2.0_r64*me/amu !Rydberg mass to amu
-  real(r64), parameter :: bohr2nm=0.052917721092_r64 !Bohr to nm
+  real(r64), parameter :: bohr2nm = 0.052917721092_r64 !Bohr to nm
   
   !Miscellaneous
   complex(r64), parameter :: oneI = (0.0_r64,1.0_r64)
   complex(r64), parameter :: twopiI = twopi*oneI
 
-  !The code below is adapted from ShengBTE (file data.f90):
-  
-  !Periodic table from ShengBTE
-  character(len = 3), parameter :: periodic_table(114)=[character(len=3) :: &
-       "H","He","Li","Be","B","C","N","O", &
-       "F","Ne","Na","Mg","Al","Si","P","S","Cl","Ar","K","Ca","Sc","Ti","V", &
-       "Cr","Mn","Fe","Co","Ni","Cu","Zn","Ga","Ge","As","Se","Br","Kr","Rb", &
-       "Sr","Y","Zr","Nb","Mo","Tc","Ru","Rh","Pd","Ag","Cd","In","Sn","Sb", &
-       "Te","I","Xe","Cs","Ba","La","Ce","Pr","Nd","Pm","Sm","Eu","Gd","Tb", &
-       "Dy","Ho","Er","Tm","Yb","Lu","Hf","Ta","W","Re","Os","Ir","Pt","Au","Hg", &
-       "Tl","Pb","Bi","Po","At","Rn","Fr","Ra","Ac","Th","Pa","U","Np","Pu","Am", &
-       "Cm","Bk","Cf","Es","Fm","Md","No","Lr","Rf","Db","Sg","Bh","Hs","Mt","Ds" ,&
-       "Rg","Cn","Uuq","Uuh"]
+  !Not parameters, but effectively behaves as parameters
+  logical, private :: periodic_table_exists = .false.
+  type(fhash_tbl_t), private :: periodic_table
+
+contains
+
+  function lookup_periodic_table(element) result(data)
+      character(*), intent(in) :: element
+      type(isotopes) :: data
+
+      class(*), allocatable :: raw_data
+
+      if (.not. periodic_table_exists) then
+         call create_periodic_table(periodic_table)
+         periodic_table_exists = .true.
+      end if
+
+      call periodic_table%get_raw(key(element), raw_data)
+
+      select type (raw_data)
+      type is (isotopes)
+         data = raw_data
+      class default
+         print *, 'Error in reading periodic table for element ' // element // '.'
+      end select
+   end function lookup_periodic_table
+
+   subroutine create_periodic_table(periodic_table)
+      type(fhash_tbl_t), intent(out) :: periodic_table
```

Comment:
Brief documentation would be good.

Reply:
See earlier comment regarding having a unit test for `lookup_periodic_table`. That will implicitly test this subroutine.